crates/nix_rs/src/flake/schema.rs

```diff
@@ -1,109 +1,206 @@
-            devshells: pop_per_system_tree(output, "devShells"),
-            checks: pop_per_system_tree(output, "checks"),
-            apps: pop_per_system_tree(output, "apps"),
-            formatter: pop_leaf_type(output, "formatter"),
-            nixos_configurations: pop_tree(
-                &mut output.inventory.get_mut("nixosConfigurations"),
-                &["children"],
-            ),
-            darwin_configurations: pop_tree(
-                &mut output.inventory.get_mut("darwinConfigurations"),
-                &["children"],
-            ),
-            home_configurations: pop_tree(
-                &mut output.inventory.get_mut("homeConfigurations"),
-                &["children"],
-            ),
-            nixos_modules: pop_tree(&mut output.inventory.get_mut("nixosModules"), &["children"]),
-            docker_images: pop_tree(&mut output.inventory.get_mut("dockerImages"), &["children"]),
-            overlays: pop_tree(&mut output.inventory.get_mut("overlays"), &["children"]),
-            templates: pop_tree(&mut output.inventory.get_mut("templates"), &["children"]),
-            schemas: pop_tree(&mut output.inventory.get_mut("schemas"), &["children"]),
-            other: Some(output.clone()),
+/// A terminal value of a flake output
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Val {
+    #[serde(rename = "what")]
+    /// Represents the type of the flake output
+    pub type_: Type,
+    /// If the flake output is a derivation, this will be the name of the derivation
+    pub derivation_name: Option<String>,
+    /// A short description derived from `meta.description` of the derivation with [Val::derivation_name]
+    pub short_description: Option<String>,
+}
+
+impl Default for Val {
+    fn default() -> Self {
+        Self {
+            type_: Type::Unknown,
+            derivation_name: None,
+            short_description: None,
         }
     }
 }
+
+/// The type of a flake output [Val]
+///
+/// These types can differ based on [static@DEFAULT_FLAKE_SCHEMAS].
+/// The types here are based on <https://github.com/DeterminateSystems/flake-schemas>
+/// For example, see [NixosModule type](https://github.com/DeterminateSystems/flake-schemas/blob/0a5c42297d870156d9c57d8f99e476b738dcd982/flake.nix#L268)
+#[allow(missing_docs)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Type {
+    #[serde(rename = "NixOS module")]
+    NixosModule,
+    #[serde(rename = "NixOS configuration")]
+    NixosConfiguration,
+    #[serde(rename = "nix-darwin configuration")]
+    DarwinConfiguration,
+    #[serde(rename = "package")]
+    Package,
+    #[serde(rename = "development environment")]
+    DevShell,
+    #[serde(rename = "CI test")]
+    Check,
+    #[serde(rename = "app")]
+    App,
+    #[serde(rename = "template")]
+    Template,
+    #[serde(other)]
+    Unknown,
```

Comment:
Perhaps this should be `Other(String)`?  Assuming flake-schemas passes the arbitrary type string, that is. Not sure.

Reply:
Can address this in different PR. I'm merging this.